### PR TITLE
Couple HydroChrono and add floater classes

### PR DESCRIPTION
**Pull request by tridelat**
This PR couples HydroChrono to SEAHOWL and adds corresponding floater classes.
The implementation is tested using the IEA15MW turbine with VolturnUS-S floater with HydroChrono commit a294a34. 
Main changes:
- [x] "hydro" namespace
- [x] Coupling with HydroChrono
- [x] TurbineFloating class
- [x] Floater classes
- [x] Cleaning
- [x] ~~Test~~ _not adding a test right now as it would need pushing a binary .h5 file (tested locally)_

To potentially do:
- [x] Separate elasto and hydro parts of floater (but it adds complexity to do so) or decide to keep hydro as dependent on elasto namespace.
- [x] Make TurbineFloating elasto functions (assemble, translate, rotate) inherited from ComponentElasto (or ComponentDynamic)

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -140,6 +141,9 @@ int main(int argc, char* argv[]) {
     // statics
     if (statics_prestep) {
         system_elasto->do_statics(true, 10);
+        // for (auto& turbine : system_core.turbines) {
```
Comment (wenchaoyu) [inline src/main.cpp]: these three lines can be removed

File: src/main.cpp [hunk 2]
```diff
@@ -181,6 +186,12 @@ int main(int argc, char* argv[]) {
             application->EndScene();
 #endif
             time_outputs += dt_outputs;
+
+            if (system_core.get_time() > 3) {
+                for (auto& turbine : system_core.turbines) {
+                    turbine->tower.elasto.nodes.front()->set_fixed(false);
```
Comment (wenchaoyu) [inline src/main.cpp]: Can add a option in main.json or turbine.json to switch between bottom-fixed and floating

**Review by wenchaoyu — COMMENTED** [review]
some minor changes needed

File: src/main.cpp (hunk 1 above)
Comment (tridelat) [reply src/main.cpp]: done

File: src/main.cpp (hunk 2 above)
Comment (tridelat) [reply src/main.cpp]: good catch, this is now done automatically (the node is fixed if no floater is defined for the turbine, or free if a floater is defined)

**Review by wenchaoyu — APPROVED** [review]
ok for me 👍

**Pull request merged**